scripts/rq3-4.py

```diff
@@ -0,0 +1,238 @@
+    row ={}
+    if key == 'accuracy':
+      row['class'] = key
+      row['precision'] = ''
+      row['recall'] = ''
+      row['f1_score'] = ''
+      row['support'] = ''
+      row['classifier'] = name
+      row['accuracy'] = report[key]
+    else:
+      row['class'] = key
+      row['precision'] = report[key]['precision']
+      row['recall'] = report[key]['recall']
+      row['f1_score'] = report[key]['f1-score']
+      row['support'] = report[key]['support']
+      row['classifier'] = name
+      row['accuracy'] = ''
+
+    report_data.append(row)
+
+  dataframe = pd.DataFrame.from_dict(report_data)
+  dataframe.to_csv(f'{path_result}classification_report_{text}_{name}.csv', index = False)
+
+def classify(train, test, annotated, embedding_field, label_field, path_result):
+    names = [
+        "Logistic Regression",
+        "Nearest Neighbors",
+        "Linear SVM",
+        "RBF SVM",
+        "Decision Tree",
+        "Random Forest",
+        "AdaBoost",
+        "Naive Bayes",
+    ]
+
+    penalty = 'l2'
+    C = 1.0
+    class_weight = 'balanced'
+    random_state = 2018
+    solver = 'liblinear'
+    classifiers = [
+        LogisticRegression(penalty=penalty,
+                           C=C,
+                           class_weight=class_weight,
+                           random_state=random_state,
+                           solver=solver,
+                           ),
+
+        KNeighborsClassifier(3),
+        SVC(kernel="linear", C=0.025),
+        SVC(gamma=2, C=1),
+        DecisionTreeClassifier(max_depth=5),
+        RandomForestClassifier(max_depth=5, n_estimators=10, max_features=1),
+        AdaBoostClassifier(),
+        GaussianNB(),
+    ]
+
+    X_train = list(train[embedding_field])
+    X_test = list(test[embedding_field])
+    y_test = list(test[label_field])
+    y_train = list(train[label_field])
+    X_annotated = list(annotated[embedding_field])
+    y_annotated = list(annotated[label_field])
+
+
+    for name, clf in zip(names, classifiers):
+        try:
+            clf.fit(X_train, y_train)
+            score = clf.score(X_test, y_test)
+
+            print(score)
+
+            preds = clf.predict(X_test)
+            report = classification_report(y_test, preds, digits=6, output_dict=True)
+            name = name.replace(' ', '-')
+
+            classification_report_csv(report, 'test', name, path_result)
+
+            annot_preds = clf.predict(X_annotated)
+
+            report = classification_report(y_annotated, annot_preds, digits=6, output_dict=True)
+
+            classification_report_csv(report, 'annotated', name, path_result)
+
+            joblib.dump(clf, f'{path_result}{name}.pkl')
+        except Exception as e:
+            print(e)
+
+
+def main():
+    path_data = './../data/'
+    path_result = './../results/'
+    test_path = f'{path_data}waseem/test.txt'
+    test_path_label = f'{path_data}waseem/testGold.txt'
+    train_path = f'{path_data}waseem/wtrain.txt'
```

Comment:
typo: `f'{path_data}waseem/train.txt'`

Reply:
also `train_path_label = f'{path_data}waseem/trainGold.txt'`

Reply:
I don't think this file exists `f'{path_data}waseem/labels.txt'` If you want to use it can you add it under the `data` dir?

Reply:
This file also doesn't exist: `annotated_path = f'{path_data}annotated.txt'`